From 46bbd28d43438341e6f91f3cc582f6cd029ac713 Mon Sep 17 00:00:00 2001
From: Pontus Fuchs <pontus.fuchs@gmail.com>
Date: Thu, 23 Aug 2012 09:21:01 +0200
Subject: [PATCH 01/68] ar5523: Don't store private data in skb->cb

skb->cb is used by mac80211 and should not be used by the driver.
Move tx private data to driver_data in ieee80211_tx_info.
---
 ar5523.c |   10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

--- a/drivers/net/wireless/ar5523.c
+++ b/drivers/net/wireless/ar5523.c
@@ -99,7 +99,6 @@ struct ar5523_rx_cmd {
 
 struct ar5523_tx_data {
 	struct ar5523		*ar;
-	struct sk_buff		*skb;
 };
 
 struct ar5523_rx_data {
@@ -770,10 +769,15 @@ static int ar5523_set_rts_threshold(stru
 	return ar5523_write_reg(ar, 0x09, value);
 }
 
+static inline struct ar5523_tx_data *ar5523_get_tx_priv(struct sk_buff *skb)
+{
+	return (struct ar5523_tx_data*) &IEEE80211_SKB_CB(skb)->driver_data;
+}
+
 static void ar5523_data_tx_cb(struct urb *urb)
 {
 	struct sk_buff *skb = urb->context;
-	struct ar5523_tx_data *data = (struct ar5523_tx_data *)skb->cb;
+	struct ar5523_tx_data *data = ar5523_get_tx_priv(skb);
 	struct ar5523 *ar = data->ar;
 	struct ieee80211_tx_info *txi;
 
@@ -821,7 +825,7 @@ static void ar5523_tx(struct ieee80211_h
 		goto out_free_skb;
 	urb->context = skb;
 	
-	data = (struct ar5523_tx_data *)skb->cb;
+	data = ar5523_get_tx_priv(skb);
 	data->ar = ar;
 
 	desc = (struct ar5523_tx_desc *)skb_push(skb, sizeof(*desc));
